code/cfile/cfile.cpp

```diff
@@ -969,6 +969,7 @@ static CFILE *cf_open_fill_cfblock(const char* source, int line, FILE *fp, int t
 		cfp->dir_type = type;
 		cfp->max_read_len = 0;
 
+		cfp->original_filename = original_filename;
```

Comment:
`original_filename` points to a temporary location or a location that will be overwritten. Need to `vm_strdup()` this in all of the `cf_open_*()` functions and free it on `cfclose()` to avoid memory errors and/or invalid values when checking the value later.

Reply:
It would also be possible to have `original_filename` be an `SCP_string` since `CFILE` does not need to be trivially copyable.

Reply:
Oh good catch.